controllers/datadogagent/feature/apm/feature.go

```diff
@@ -147,12 +150,19 @@ func (f *apmFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.Re
 func (f *apmFeature) ManageDependencies(managers feature.ResourceManagers, components feature.RequiredComponents) error {
 	// agent local service
 	if component.ShouldCreateAgentLocalService(managers.Store().GetVersionInfo(), f.forceEnableLocalService) {
+		if f.hostPortEnabled {
```

Comment:
Should we check if hostNetwork is being used?

Reply:
yes if hostPortEnabled and hostNetwork are enabled the container's Pod `Port` and so `TargetPort` must be equal to the `hostPort`.
So you need also to have the `ServicePort.TargetPort` equal to `hostPort`